tests/population/test_mortality.py

```diff
@@ -53,27 +49,40 @@ def test_mortality_default_lookup_configuration(setup_sim_with_pop_and_mortality
     assert (lookup_tables["life_expectancy"].data["value"] == 98.0).all()
 
 
-def test_mortality_creates_columns(setup_sim_with_pop_and_mortality):
+def test_mortality_creates_attributes(setup_sim_with_pop_and_mortality):
     sim, bp, mortality = setup_sim_with_pop_and_mortality
     pop = sim.get_population()
     expected_columns_created = set(mortality.columns_created)
-    mortality_created_columns = set(pop.columns).difference(
-        set(bp.columns_created + ["tracked"])
+    expected_attributes_created = {
+        mortality.cause_specific_mortality_rate_pipeline_name,
+        mortality.unmodeled_csmr_pipeline_name,
+        mortality.unmodeled_csmr_paf_pipeline_name,
+    }
+    # BasePopulation and AgedOutSimulants create columns themselves
+    other_columns_created = set(bp.columns_created + ["is_aged_out"])
+    mortality_created_columns = set(pop.columns.get_level_values(0)).difference(
+        other_columns_created
+    )
+    # sets do not guarantee order so we assert te difference is empty
+    assert (
+        expected_columns_created.union(expected_attributes_created).difference(
+            mortality_created_columns
+        )
+        == set()
     )
```

Comment:
Since sets are unordered, the order doesn't matter and you can just do:
```python
assert expected_columns_created.union(expected_attributes_created) == mortality_created_columns
```